packages/cdk-ecs-codedeploy-resources/lib/index.ts

```diff
@@ -0,0 +1,268 @@
+
+    var terminationTimeout: number = props.stage.includes('prod') ? 120 : 0;
+
+    const bucket = new s3.Bucket(this, 'Bucket', {
+      bucketName: `deploy-${props.appName}.${props.stage}.ecs`,
+      publicReadAccess: true,
+      versioned: true,
+    });
+
+    const originAccessIdentity = new cloudfront.OriginAccessIdentity(this, 'OriginAcessIdentity', {
+      comment: `OriginAccessIdentity for ${bucket.bucketName}.`
+    });
+
+    const bucketPolicy = new s3.BucketPolicy(this, 'BucketPolicy', {
+      bucket: bucket
+    });
+
+    bucketPolicy.document.addStatements(
+      new iam.PolicyStatement({
+        effect: iam.Effect.ALLOW,
+        principals: [originAccessIdentity.grantPrincipal],
+        actions: ['s3:GetObject'],
+        resources: [bucket.bucketArn + "/*"],
+      })
+    );
+
+    let output = new cdk.CfnOutput(this, "AppspecBucket", {
+      value: bucket.bucketName
+    });
+
+    let output2 = new cdk.CfnOutput(this, "OriginAccessIdentity", {
+      value: originAccessIdentity.originAccessIdentityName
+    });
+
+    output.overrideLogicalId("AppspecBucket");
+    output2.overrideLogicalId("OriginAccessIdentity");
+
+    let codedeployCreateApplicationCall: customresource.AwsSdkCall = {
+      service: 'CodeDeploy',
+      action: 'createApplication',
+      parameters: {
+        applicationName: `${props.stage}-${props.appName}-CodeDeployApplication`,
+        computePlatform: "ECS"
+      },
+      physicalResourceId: customresource.PhysicalResourceId.of('deployment-application-custom-resource')
+    }
+
+    let codedeployUpdateApplicationCall: customresource.AwsSdkCall = {
+      service: 'CodeDeploy',
+      action: 'updateApplication',
+      parameters: {
+        applicationName: `${props.stage}-${props.appName}-CodeDeployApplication`
+      },
+      physicalResourceId: customresource.PhysicalResourceId.of('deployment-application-update-custom-resource')
+    }
+
+    const codeDeployApplication = new customresource.AwsCustomResource(this, 'CodeDeployApplication', {
+      onCreate: codedeployCreateApplicationCall,
+      onUpdate: codedeployUpdateApplicationCall,
+      onDelete: {
+        service: 'CodeDeploy',
+        action: 'deleteApplication',
+        parameters: {
+          applicationName: `${props.stage}-${props.appName}-CodeDeployApplication`,
+        }
+      },
+      policy: customresource.AwsCustomResourcePolicy.fromSdkCalls({ resources: customresource.AwsCustomResourcePolicy.ANY_RESOURCE }),
+      role: iamRole
+    });
+
+    let codedeployCreateDeploymentGroupCall: customresource.AwsSdkCall = {
+      service: 'CodeDeploy',
+      action: 'createDeploymentGroup',
+      parameters: {
+        applicationName: `${props.stage}-${props.appName}-CodeDeployApplication`,
+        deploymentGroupName: `${props.stage}-${props.appName}-CodeDeployDeploymentGroup`,
+        serviceRoleArn: "arn:aws:iam::022393549274:role/CodeDeployServiceRoleECS",
```

Comment:
Same thing as above. This can't be hardcoded but if this role is provided by AWS just format the string with the account id.